transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/LegacyFlgDestination.java

```diff
@@ -46,22 +61,43 @@ public String getName() {
   @NonNull
   @Override
   public byte[] getExtras() {
-    return encodeString(apiKey);
+    return encodeAsExtras();
   }
 
   @NonNull
-  static byte[] encodeString(@NonNull String s) {
+  public static LegacyFlgDestination parseFromExtras(@Nullable byte[] a) {
+    if (a == null) {
+      return LegacyFlgDestination.DEFAULT_INSTANCE;
+    }
     try {
-      return s.getBytes("UTF-8");
+      String encodedExtra = new String(a, "UTF-8");
+      if (!encodedExtra.startsWith(EXTRAS_VERSION_MARKER)) {
+        throw new IllegalStateException("Version marker missing from extras");
```

Comment:
We need to make sure we don't propagate exceptions outside of the backend class, otherwise it will crash runtime

Reply:
I'd made the `CctTransportBackend` handle the exception by logging the error and not trying to send the event. Does that sounds as a reasonable approach?